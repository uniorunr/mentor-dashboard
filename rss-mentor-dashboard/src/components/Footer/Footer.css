.footer {
  position: absolute;
  bottom: 0;
  width: 100%;
  color: rgba(0, 0, 0, 0.5);
}

.footer-wrapper {
  width: 90%;
  margin: 0 auto;
}

.footer p {
  margin: 1.25em 0;
  font-family: "Roboto", "Helvetica", "Arial", sans-serif;
  font-size: 0.9em;
}

.footer .red-heart {
  color: rgba(255, 0, 0, 0.9);
}

.footer a {
  color: rgba(0, 0, 0, 0.75);
  text-decoration: none;
}

.footer a:hover {
  color: rgba(0, 0, 0, 1);
}