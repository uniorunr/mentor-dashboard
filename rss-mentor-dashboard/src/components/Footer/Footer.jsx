import React from 'react';
import PropTypes from 'prop-types';
import './Footer.css';

const Footer = ({ link, text }) => (
  <footer className="footer">
    <div className="footer-wrapper">
      <p>Developed with
        <span className="red-heart" role="img" aria-label="red heart"> ❤️ </span>
        by <a href={link} target="_blank" rel="noopener noreferrer">{text}</a>
      </p>
    </div>
  </footer>
);

Footer.propTypes = {
  link: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default Footer;
