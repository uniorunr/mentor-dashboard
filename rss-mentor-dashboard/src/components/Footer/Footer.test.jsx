import React from 'react';
import renderer from 'react-test-renderer';
import Footer from './Footer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Footer link="test" text="test" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
