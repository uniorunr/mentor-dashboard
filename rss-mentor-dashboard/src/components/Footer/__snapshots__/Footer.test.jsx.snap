// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  className="footer"
>
  <div
    className="footer-wrapper"
  >
    <p>
      Developed with
      <span
        aria-label="red heart"
        className="red-heart"
        role="img"
      >
         ❤️ 
      </span>
      by 
      <a
        href="test"
        rel="noopener noreferrer"
        target="_blank"
      >
        test
      </a>
    </p>
  </div>
</footer>
`;
